crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -1042,6 +1042,132 @@ reveal_type(A.__mro__)
 reveal_type(A.X)  # revealed: Unknown | Literal[42]
 ```
 
+## Intersections of attributes
+
+### Attribute only available on one element
+
+```py
+from knot_extensions import Intersection
+
+class A:
+    x: int = 1
+
+class B: ...
+
+def _(a_and_b: Intersection[A, B]):
+    reveal_type(a_and_b.x)  # revealed: int
+
+# Same for class objects
+def _(a_and_b: Intersection[type[A], type[B]]):
```

Comment:
I wonder if we should allow `type[Intersection[A, B]]` in type expressions, as a shorthand for `Intersection[type[A], type[B]]`. We already allow (and it's managed by the spec) `type[A | B]` and `type[Union[A, B]]` as shorthands for `type[A] | type[B]`.

Obviously doesn't need to be done in this PR, though